src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,170 @@
+    {
+        return $this->client->setMulti($values, $lifetime)
+            && $this->isPreviousClientActionSuccessful();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->client->getMulti($ids) as $id => $val) {
+            yield $id => $val;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
+            && $this->client->getResultCode() !== \Memcached::RES_NOTFOUND;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids)
+    {
+        $toDelete = count($ids);
+        foreach ((array) $this->client->deleteMulti($ids) as $result) {
+            if (true === $result || \Memcached::RES_NOTFOUND === $result) {
+                --$toDelete;
+            }
+        }
+
+        return 0 === $toDelete;
+    }
+
+    protected function addServer($dsn)
+    {
+        list($host, $port, $weight) = $this->dsnExtract($dsn);
+
+        return $this->isServerInClientPool($host, $port)
+            || ($this->client->addServer($host, $port, $weight)
+                && $this->isPreviousClientActionSuccessful());
+    }
+
+    protected function setOption($opt, $val)
+    {
+        list($opt, $val) = $this->optionSanitize($opt, $val);
+
+        $restore = error_reporting(~E_ALL);
+        $success = $this->client->setOption($opt, $val);
+        error_reporting($restore);
+
+        return $success && $this->isPreviousClientActionSuccessful();
+    }
+
+    protected function getAllIdsMatching($namespace)
+    {
+        if (false === $ids = $this->client->getAllKeys()) {
+            return false;
+        }
+
+        return array_filter((array) $ids, function ($id) use ($namespace) {
+            return 0 === strpos($id, $namespace);
+        });
+    }
+
+    private function optionSanitize($opt, $val)
+    {
+        if (false === filter_var($opt = $this->optionResolve($opt), FILTER_VALIDATE_INT)) {
+            throw new InvalidArgumentException(sprintf('Invalid Memcached option type: %s (expects an int or a resolvable client constant)', $opt));
+        }
+
+        if (false === filter_var($val = $this->optionResolve($val), FILTER_VALIDATE_INT) &&
+            null  === filter_var($val, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE)) {
+            throw new InvalidArgumentException(sprintf('Invalid Memcached option value: %s (expects an int, a bool, or a resolvable client constant)', $val));
+        }
+
+        return array($opt, $val);
+    }
+
+    private function optionResolve($val)
+    {
+        return defined($constant = '\Memcached::'.strtoupper($val)) ? constant($constant) : $val;
+    }
+
+    private function isPreviousClientActionSuccessful()
+    {
+        return $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    protected function isServerInClientPool($host, $port)
```

Comment:
This is majorly important depending on the environment PHP is running in. For example, given the following service definition

```yml
my.cache:
  class: Symfony\Component\Cache\Adapter\MemcachedAdapter
  calls:
    [setup, [["%memcached_dsn_01%", "%memcached_dsn_01%"]]
```

On every page request, the registered servers are actually persisted (again, depending on environment), meaning after 100 page requests you'll have 200 servers registered with the extension.